samples/meta_reader/main.cpp

```diff
@@ -47,6 +66,75 @@ struct writer : writer_base<writer>
         return generic_param_guard{ this };
     }
 
+    void write_indent()
+    {
+        for (int32_t i = 0; i < indent; i++)
+        {
+            writer_base::write_impl("    ");
+        }
+    }
+
+    void write_impl(std::string_view const& value)
+    {
+        std::string_view::size_type current_pos{ 0 };
+        auto on_new_line = back() == '\n';
+
+        while (true)
+        {
+            auto pos = value.find('\n', current_pos);
+
+            if (pos == std::string_view::npos)
+            {
+                if (current_pos < value.size())
+                {
+                    if (on_new_line)
+                    {
+                        write_indent();
+                    }
+
+                    writer_base::write_impl(value.substr(current_pos));
+                }
+
+                return;
+            }
+
+            auto current_line = value.substr(current_pos, pos - current_pos + 1);
+            auto empty_line = current_line[0] == '\n';
+
+            if (on_new_line && !empty_line) {
+                write_indent();
+            }
+
+            writer_base::write_impl(current_line);
+
+            on_new_line = true;
+            current_pos = pos + 1;
+        }
+    }
+
+    void write_impl(char const value)
+    {
+        if (back() == '\n' && value != '\n')
+        {
+            write_indent();
+        }
+
+        writer_base::write_impl(value);
+    }
+
+    template <typename... Args>
+    std::string write_temp(std::string_view const& value, Args const& ... args)
```

Comment:
pass `string_view` by value

Reply:
Not fixed.
Base class struct writer_base at src/library/text_writer has string_view passed by const reference. I just copied implementation from src/tool/python/type_writers.h. And all other code in this repository always pass string_views as const references and not by copy/value. I prefer it too, since it is more efficient to pass 64-bit pointer in one register, rather than use two registers for pointer and size. Especially if same string_view is passed down multiple functions deep.

Reply:
Yep, consistency trumps here. The xlang repo uses `const&` wherever possible. It is also more efficient as you mentioned.